src/Database/Schema/CachedCollection.php

```diff
@@ -77,7 +77,12 @@ public function describe($name, array $options = [])
      */
     public function cacheKey($name)
     {
-        return $this->_connection->configName() . '_' . $name;
+        $cachePrefix = $this->_connection->configName();
+        $config = $this->_connection->config();
+        if (isset($config['cacheKeyPrefix'])) {
+            $cachePrefix = $config['cacheKeyPrefix'];
+        }
+        return $cachePrefix . '_' . $name;
```

Comment:
I would consider a "true" prefix to include the `_` here, no magic internal additions IMO. Or is this how it works in 4.x now as well?

Reply:
I do not understand that. It is `return $this->prefix . '_' . $name;` in https://github.com/cakephp/cakephp/blob/4.x/src/Database/Schema/CachedCollection.php#L98.

Reply:
I see.
I would like to get feedback here from @ADmad @markstory etc on if we can change 4.x here also to make prefixing behave as in other cases - and how intuitively expected (to not magically append another glue char.

Reply:
I am fine with not auto adding `_` if prefix has been explicitly specified if someone wants to update 4.x.

Reply:
But I see also
- https://github.com/cakephp/cakephp/blob/69643f58dd110a4c5cce7ff867707de176c64bc4/src/View/Helper/IdGeneratorTrait.php#L98

which is in line with this change however. Even though that is for HTML DOM - and not as file, path or cache prefix.

Reply:
The doc block on 4.x states

 > Typically the connection name.

As such it is probably fine to keep it as is (and auto append the underscore).

Reply:
I don't see a problem with the underscore. It is a consistent input to the cache key and makes keys more readable should you need to inspect them.